src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,10 +76,19 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
-        if (!isValidAddress(address)) {
+        String[] addressParts = splitAddress(address);
+        if (!isValidAddress(address) || addressParts.length != ADDRESS_SIZE) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        //this.value = address;
+        block = new Block(addressParts[0]);
+        street = new Street(addressParts[1]);
+        unit = new Unit(addressParts[2]);
+        postalCode = new PostalCode(addressParts[3]);
+    }
+    
+    public String[] splitAddress(String address) {
+        return address.split(", ");
```

Comment:
doesn't this split fail for addresses with no space after comma, like: `123,some road,#12-12,s123456`